### ISSUE TITLE
Istio images version mismatch after names change

### ISSUE THREAD

**Issue by mjakobczyk** [kyma-incubator/reconciler#744]
**Description**

Istio reconciliation fails after images' offered by SoloIO team changed their names.

**Expected result**

No version mismatch, as this one is expected. Implemented changes need to support all versions of managed kyma (e.g. 2, main)

**Actual result**

```bash
2022-01-31T22:50:01.838Z	DEBUG	service/runner.go:132	Runner: Reconcile action of 'istio-configuration' with version 'local' failed: Istio components version mismatch detected: pilot version: 1.12.2, data plane version: 1.12.2-solo-fips
```

**Steps to reproduce**

Changes introduced by [this PR](https://github.com/kyma-project/kyma/pull/13229/files) popped this up.

**Troubleshooting**

Have a look at the following Reconciler code:

`pkg/reconciler/instances/istio/action.go:369`
```go
func isMismatchPresent(istioStatus actions.IstioStatus) bool {
	pilot := newHelperVersionFrom(istioStatus.PilotVersion)
	dataPlane := newHelperVersionFrom(istioStatus.DataPlaneVersion)
	return pilot.compare(&dataPlane) != 0
}
```

Then this mismatch is being used here:

`pkg/reconciler/instances/istio/action.go:105`
```go
	if isMismatchPresent(istioStatus) {
		errorMessage := fmt.Sprintf("Istio components version mismatch detected: pilot version: %s, data plane version: %s", istioStatus.PilotVersion, istioStatus.DataPlaneVersion)
		return errors.New(errorMessage)
	}
```

Consider logging a warning it instead of throwing an error.

**Links**
Found during: https://github.com/kyma-project/kyma/pull/13229

**Comment by strekm** [comment]
Bug fix needs mandatory tests agains kyma 2.0 and main

**Comment by mjakobczyk** [comment]
Tested the solution again against both Kyma `main` and `release-2.0` branches. I used istioctl in version 1.11.4 for testing purposes.

Results for branch `main`:
- command:
```bash
./bin/mothership-darwin local --value global.ingress.domainName=example.com,global.domainName=example.com --components istio-configuration --version main
```
- output:
```log
{"level":"INFO","time":"2022-02-17T12:41:28.573+0100","message":"Component 'istio-configuration' has status 'running'"}
{"level":"INFO","time":"2022-02-17T12:41:29.335+0100","message":"Resolved target Istio version: 1.11.4-solo-fips-distroless from values"}
```

- summary: Istio target version resolved from values as expected ✅ 

Results for branch `release-2.0`:
- command:
```bash
./bin/mothership-darwin local --value global.ingress.domainName=example.com,global.domainName=example.com --components istio-configuration --version release-2.0
```
- output:
```log
{"level":"INFO","time":"2022-02-17T12:46:21.922+0100","message":"Component 'istio-configuration' has status 'running'"}
{"level":"INFO","time":"2022-02-17T12:46:22.671+0100","message":"Resolved target Istio version: 1.11.2 from Chart definition"}
```

- summary: Istio target version resolved from Chart definition as expected ✅

**Comment by mjakobczyk** [comment]
We encounter a problem of version mismatch during bump to Istio 1.12.2. I checked how `istioctl` reports versions for each of Istio revision we currently use.
- for Kyma `release-2.0` branch there is no version mismatch (data and control planes are the same): ✅ 
```bash
> istioctl version
client version: 1.11.2
control plane version: 1.11.2
data plane version: 1.11.2 (1 proxies)
```

- for Kyma `main` branch there is no version mismatch (data and control planes are the same): ✅ 
```bash
istioctl version
client version: 1.11.4
control plane version: 1.11.4-solo-fips
data plane version: 1.11.4-solo-fips (1 proxies)
```

- for Kyma `PR-13229` branch: (which includes a mentioned bump) there is a version mismatch (data and control planes are reported differently): 🚫 
```bash
client version: 1.12.2
control plane version: 1.12.2
data plane version: 1.12.2-solo-fips (1 proxies)
```

The comparison between `1.12.2` and `1.12.2-solo-fips` simply returns false as they are semantically different. The mechanism that I introduced is a `helperVersion` which under the hood covers only major, minor and patch versions with no additional data. As a result, it required refactoring which is present in PR #845 .

The testing results are as it follows:
- ...
